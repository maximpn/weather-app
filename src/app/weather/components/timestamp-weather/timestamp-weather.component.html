<div class="date-time-wrapper">
  <time class="time">{{ weather.dt * 1000 | date:'H:mm' }}</time>
  <time class="date">{{ weather.dt * 1000 | date:'MMM d' }}</time>
</div>

<wa-weather-condition-icon [weatherCondition]="weather.weather[0]" size="2x"></wa-weather-condition-icon>

<span class="temperature">{{ weather.temp | number:'1.0-0' }}°</span>
<span class="real-feel">Feels like: {{ weather.feels_like | number:'1.0-0' }}°</span>
<span class="phrase">{{ weather.weather[0].description }}</span>
<div class="clouds">
  <mat-icon>cloud_queue</mat-icon>
  <span class="value">{{ weather.clouds }}%</span>
</div>
