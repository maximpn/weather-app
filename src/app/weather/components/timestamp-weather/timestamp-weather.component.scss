:host {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0 20px;
}

.date-time-wrapper {
  width: 80px;
  display: flex;
  flex-direction: column;
  align-items: center;

  .time {
    font-size: 32px;
  }

  .date {
    margin-top: 5px;
  }
}

.temperature {
  font-size: 48px;
}

.real-feel {
  color: #878787;
}

.phrase {
  width: 200px;
  text-align: center;
  text-transform: capitalize;
}

.clouds {
  display: flex;

  .value {
    line-height: 24px;
    margin-left: 10px;
  }
}
