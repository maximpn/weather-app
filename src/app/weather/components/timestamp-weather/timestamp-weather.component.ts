import { Component, Input } from '@angular/core';

import { TimestampWeather } from '../../models/timestamp-weather.model';

@Component({
  selector: 'wa-timestamp-weather',
  templateUrl: 'timestamp-weather.component.html',
  styleUrls: ['timestamp-weather.component.scss'],
})
export class TimestampWeatherComponent {
  @Input()
  weather!: TimestampWeather;
}
