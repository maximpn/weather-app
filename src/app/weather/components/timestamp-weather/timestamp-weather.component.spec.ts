import { Component, Input, NO_ERRORS_SCHEMA } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';

import { TimestampWeather } from '../../models/timestamp-weather.model';
import { WeatherCondition } from '../../models/weather-condition.model';

import { TimestampWeatherComponent } from './timestamp-weather.component';

@Component({
  selector: 'wa-weather-condition-icon',
  template: '{{ weatherCondition | json }}',
})
class WeatherConditionIconMockComponent {
  @Input()
  weatherCondition!: WeatherCondition;
}

describe('TimestampWeatherComponent', () => {
  let fixture: ComponentFixture<TimestampWeatherComponent>;

  beforeEach(() => {
    fixture = TestBed.configureTestingModule({
      declarations: [
        TimestampWeatherComponent,
        WeatherConditionIconMockComponent,
      ],
      schemas: [
        NO_ERRORS_SCHEMA,
      ],
    }).createComponent(TimestampWeatherComponent);

    fixture.componentInstance.weather = {
      dt: new Date(2020, 5, 2).getTime() / 1000,
      weather: [
        { id: 1, main: 'cloudy', description: 'cloudy', icon: '02d' },
      ],
      temp: 30.55,
      feels_like: 31,
      clouds: 20,
    } as TimestampWeather;

    fixture.detectChanges();
  });

  it('should render the component', () => {
    expect(fixture.nativeElement).toMatchSnapshot();
  });
});
