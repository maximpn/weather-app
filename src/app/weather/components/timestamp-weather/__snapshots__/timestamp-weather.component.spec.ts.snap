// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimestampWeatherComponent should render the component 1`] = `
<div
  id="root0"
>
  <div
    class="date-time-wrapper"
  >
    <time
      class="time"
    >
      0:00
    </time>
    <time
      class="date"
    >
      Jun 2
    </time>
  </div>
  <wa-weather-condition-icon
    size="2x"
  >
    {
  "id": 1,
  "main": "cloudy",
  "description": "cloudy",
  "icon": "02d"
}
  </wa-weather-condition-icon>
  <span
    class="temperature"
  >
    31°
  </span>
  <span
    class="real-feel"
  >
    Feels like: 31°
  </span>
  <span
    class="phrase"
  >
    cloudy
  </span>
  <div
    class="clouds"
  >
    <mat-icon>
      cloud_queue
    </mat-icon>
    <span
      class="value"
    >
      20%
    </span>
  </div>
</div>
`;
